internal/cli/root.go

```diff
@@ -70,7 +85,7 @@ func NewClient(consumerToken, apiURL, Version string) *Client {
 }
 
 func (c *Client) apiConnect() error {
-	client, err := packngo.NewClientWithBaseURL(c.consumerToken, c.metalToken, nil, c.apiURL)
+	client, err := packngo.NewClientWithBaseURL(c.consumerToken, c.metalToken, c.httpClient, c.apiURL)
```

Comment:
I think we could reduce the plumbing a little if the headers were stored in the `Client` during `Config` rather than an `httpClient`. 

Storing the client in the config bothers at the surface level even though we are already doing it with the packngo.Client, but especially because we would be storing the httpClient in the config twice (within the packngo.Client and separately).


`Client.API` which is passed `cmd`, could
- if c.apiClient == nil, call `getAdditionalHeaders`
- create the httpClient
- pass it as an argument to apiConnect